index.js

```diff
@@ -26,6 +26,18 @@ module.exports = function () {
 		return str;
 	}
 
+	var isLastCharLower = false;
+	for (var i = 0; i < str.length; i++) {
+		var c = str.charAt(i);
+		if (isLastCharLower && c.toUpperCase() === c) {
+			str = str.substr(0, i) + '-' + str.substr(i);
+			isLastCharLower = false;
+			i++;
+		} else {
+			isLastCharLower = (c.toLowerCase() === c);
+		}
+	}
+
```

Comment:
Ah, good point. I guess `FBBazzy-style` => `fbbazzyStyle` is fine then.

I also think https://github.com/sindresorhus/camelcase/blob/b4c5c8d65d01576496eefe5132276eee409029ce/test.js#L25 is incorrect. `fooBar-baz` should become `fooBarBaz`, not `foobarBaz`. It should try to preserve existing camelcasing whenever possible.